src/language/templates/ast/ast.hpp

```diff
@@ -258,7 +255,7 @@ namespace ast {
          *
          * \sa Ast::get_node_type_name Ast::get_node_name
          */
-        {{ virtual(node) }}void set_name(std::string name){{ override(node) }} {
+        {{ virtual(node) }}void set_name(std::string name) {
```

Comment:
> override(node) is a macro to add override only where needed and
> copied from virtual(node). However, override is required also
> for derived classes were virtual is not needed (we do not want
> a virtual table). It was used only for get_token and set_name,
> 2 methods that are in ast_commons and, thus, must be always
> overridden.

yes, agree!

But, could you clarify why `override` is missing for `set_name`?

Related problem that I encountered in #272:

* `name` is quite common member variable in AST classes and we generate setter method `set_name` for the same
*  this results into `void set_name(std::shared_ptr<Identifier>&& name)` method in many classes of AST hierarchy.
* because of problem you mentioned above, we don't have "correct" override for methods like set_name

This results into large number of warnings : 

```bash
$ /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/compilers/2020-02-01/linux-rhel7-x86_64/gcc-4.8.5/pgi-19.4-hdirysdrvd/linux86-64-llvm/19.4/bin/pgc++  -DEIGEN_DONT_VECTORIZE=1 -DJSON_SKIP_UNSUPPORTED_COMPILER_CHECK=1 -I/gpfs/bbp.cscs.ch/apps/tools/install/linux-rhel7-x86_64/gcc-4.8.5/flex-2.6.3-r2qhjn/include -I/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl -I/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/src -I/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src -I/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/ext -I/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/ext/fmt/include -I/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/ext/spdlog/include -I/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/ext/cli11/include  -fPIC   --c++11 -o CMakeFiles/lexer_obj.dir/__/ast/ast.cpp.o -c /gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/ast/ast.cpp
"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 2727: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::IndexedName::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 2734: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::IndexedName::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 2980: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::VarName::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 2987: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::VarName::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 3240: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Argument::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 3247: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Argument::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 3490: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ReactVarName::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<VarName>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 3497: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ReactVarName::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<VarName>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 3697: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ReadIonVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 3704: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ReadIonVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 3904: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::WriteIonVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 3911: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::WriteIonVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 4111: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::NonspecificCurVar::set_name" -- virtual function
          override intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 4118: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::NonspecificCurVar::set_name" -- virtual function
          override intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 4318: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ElectrodeCurVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 4325: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ElectrodeCurVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 4525: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::SectionVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 4532: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::SectionVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 4732: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::RangeVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 4739: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::RangeVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 4939: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::GlobalVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 4946: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::GlobalVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 5146: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::PointerVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 5153: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::PointerVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 5354: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::BbcorePointerVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 5361: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::BbcorePointerVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 5561: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ExternVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 5568: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ExternVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 5768: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ThreadsafeVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 5775: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ThreadsafeVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 8491: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::DerivativeBlock::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 8498: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::DerivativeBlock::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 8795: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::LinearBlock::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 8802: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::LinearBlock::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 9111: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::NonLinearBlock::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 9118: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::NonLinearBlock::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 9403: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::DiscreteBlock::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 9410: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::DiscreteBlock::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 9680: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::PartialBlock::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 9687: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::PartialBlock::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 9970: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::FunctionTableBlock::set_name" -- virtual function
          override intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 9977: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::FunctionTableBlock::set_name" -- virtual function
          override intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 10288: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::FunctionBlock::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 10295: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::FunctionBlock::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 10621: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ProcedureBlock::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 10628: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ProcedureBlock::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 13005: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::KineticBlock::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 13012: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::KineticBlock::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 14205: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Unit::set_name" -- virtual function override intended?
                         void set_name(std::shared_ptr<String>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 14212: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Unit::set_name" -- virtual function override intended?
                         void set_name(const std::shared_ptr<String>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 14633: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::LocalVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 14640: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::LocalVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 15281: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::PlotVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 15288: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::PlotVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 15529: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ConstantVar::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 15536: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ConstantVar::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 17703: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::FunctionCall::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 17710: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::FunctionCall::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 18533: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Match::set_name" -- virtual function override intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 18540: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Match::set_name" -- virtual function override intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 19241: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::FactorDef::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 19248: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::FactorDef::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 20363: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Define::set_name" -- virtual function override intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 20370: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Define::set_name" -- virtual function override intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 20831: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ParamAssign::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 20838: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ParamAssign::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 21095: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Stepped::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 21102: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Stepped::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 21411: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::IndependentDefinition::set_name" -- virtual function
          override intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 21418: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::IndependentDefinition::set_name" -- virtual function
          override intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 21771: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::AssignedDefinition::set_name" -- virtual function
          override intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 21778: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::AssignedDefinition::set_name" -- virtual function
          override intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 22096: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::PlotDeclaration::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<PlotVar>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 22103: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::PlotDeclaration::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<PlotVar>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 23012: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::FromStatement::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 23019: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::FromStatement::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 23292: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ForAllStatement::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 23299: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::ForAllStatement::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 24865: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::PartialBoundary::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 24872: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::PartialBoundary::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 26377: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Compartment::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 26384: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Compartment::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 26640: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::LonDifuse::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 26647: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::LonDifuse::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 27218: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::LagStatement::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 27225: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::LagStatement::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 27454: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::QueueStatement::set_name" -- virtual function override
          intended?
                         void set_name(std::shared_ptr<Identifier>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 27461: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::QueueStatement::set_name" -- virtual function override
          intended?
                         void set_name(const std::shared_ptr<Identifier>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 28203: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Suffix::set_name" -- virtual function override intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 28210: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Suffix::set_name" -- virtual function override intended?
                         void set_name(const std::shared_ptr<Name>& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 28476: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Useion::set_name" -- virtual function override intended?
                         void set_name(std::shared_ptr<Name>&& name) {
                              ^

"/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/tmp/nmodl/build/src/as
          t/ast.hpp", line 28483: warning: function
          "nmodl::ast::Ast::set_name(std::__cxx11::string)" is hidden by
          "nmodl::ast::Useion::set_name" -- virtual function override intended?
                         void set_name(const std::shared_ptr<Name>& name) {
```

One solution could be to check if parent (in YAML) has the method and then add appropriate `override` annotation?